realize stride symbols in estimate_runtime

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #146752

Unfortuanlty could not create a local repo, or unit test.
fix https://github.com/pytorch/pytorch/issues/146686

laithsakka committed:
realize stride symbols in estimate_runtime

[ghstack-poisoned]

bobrenjc93 reviewed (approved):
Can't you write a unit test that calls estimate_runtime with a fabricated node? We do this type of testing a lot in dynamic shapes.

bdhirsh reviewed (approved):
did you not have any luck tweaking my example to get a repro? I also found creating a repro for my case to be annoying because the partitioner "recompute" logic only fires in certain cases, so it took some time to figure out (code: https://github.com/pytorch/pytorch/pull/144097/files#diff-4c82a5798a61d4cceb176b2700ba6fdd7c3e72d575b8e7e22458589139459caaR5778)

laithsakka commented:
> did you not have any luck tweaking my example to get a repro? I also found creating a repro for my case to be annoying because the partitioner "recompute" logic only fires in certain cases, so it took some time to figure out (code: https://github.com/pytorch/pytorch/pull/144097/files#diff-4c82a5798a61d4cceb176b2700ba6fdd7c3e72d575b8e7e22458589139459caaR5778)

I tried , but was not able to trigger  is_continuous to be called on top of it? if you have thoughts lmk

laithsakka commented:
> Can't you write a unit test that calls estimate_runtime with a fabricated node? We do this type of testing a lot in dynamic shapes.

idk how easy that would be, but if we are going that path  i can just factor out the realize logic in a function and call it realize_symbols (tensor)->tensor
and test

laithsakka committed:
Update on "realize stride symbols in estimate_runtime"


Unfortuanlty could not create a local repo, or unit test.
fix https://github.com/pytorch/pytorch/issues/146686



[ghstack-poisoned]

laithsakka committed:
Update on "realize stride symbols in estimate_runtime"


Unfortuanlty could not create a local repo, or unit test.
fix https://github.com/pytorch/pytorch/issues/146686



[ghstack-poisoned]

laithsakka committed:
Update on "realize stride symbols in estimate_runtime"


Unfortuanlty could not create a local repo, or unit test.
fix https://github.com/pytorch/pytorch/issues/146686



[ghstack-poisoned]

laithsakka commented:
add unit test

laithsakka commented:
@pytorchbot  merge

xuanzhang816 commented on torch/_functorch/partitioners.py:
```diff
@@ -1450,20 +1450,26 @@ def _optimize_runtime_with_given_memory(
+
+    shape = [realize_symbol(s) for s in shape]
+    stride = [realize_symbol(s) for s in x.stride()]
+    print(shape)
+    print(stride)
```
can we remove these prints?

bdhirsh replied:
PR'd: https://github.com/pytorch/pytorch/pull/147749
